### PR TITLE
Update composer.json to allow PHP 8

### PR DESCRIPTION
**Pull request by sanmai**

### CODE REVIEW

**Comment by martin-helmich** [comment]
Thanks for this PR! Unfortunately, after enabling the actual pipelines for this PR, I've noticed that currently Composer is not able to resolve a compatible set of dependencies for this project for PHP 8 -- mainly because there's no PHP 8-compatible version of phpspec/prophecy (already tracked at phpspec/prophecy#491). So there's no way the new CI jobs would pass with PHP 8.

I'll sit on this for a while -- hopefully, this'll resolve itself in the next few days.

Oh, and we'd also need to exclude the "PHP 8"+"PHPUnit 8" combination from the build matrix, since PHPUnit 8 is declared to be compatible with PHP `^7.2`, only.

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
   build:
     strategy:      
       matrix:
-        php: ['7.2', '7.3', '7.4']
+        php: ['7.2', '7.3', '7.4', '8.0']
         phpunit: ['8.0', '9.0']
         exclude:
```
Comment (martin-helmich) [inline .github/workflows/php.yml:17]:
```suggestion
        exclude:
          - php: '8.0'
            phpunit: '8.0'
```

**Review by martin-helmich — CHANGES_REQUESTED** [review]

**Comment by sanmai** [comment]
Thanks for the review! I made it such so failures on 8.0 would not affect the entire build. This should work, but let me try with something else too.

**Comment by sanmai** [comment]
Alright, it is as good as it gets. Hope you can squash-merge it.

I added a pretty name for the builds, you might want to update your protected branch configuration. Or we can restore the old naming scheme.

**Review by martin-helmich — APPROVED** [review]

**Comment by martin-helmich** [comment]
:+1: Looking good! Thanks for your contribution.

**Pull request merged**